### PR TITLE
Feat/96 implement pagination component

### LINKED ISSUE #96: [FEAT] External Dynamic Pages > Career Page > Implement Pagination
## Description
Develop a pagination component for the Career Page to allow users to navigate through multiple pages of job listings efficiently. The pagination should be intuitive, accessible, and integrate seamlessly with the existing job listing functionality.

## Acceptance Criteria

- The pagination component displays the current page number and adjacent page numbers.
- "Previous" and "Next" buttons are included and function correctly.
- The component updates the job listings when a new page is selected.
- The pagination is responsive and maintains usability across different screen sizes.

### Requirements	
The pagination component should include:

- Current page indicator (highlighted)
- Clickable page numbers
- "Previous" and "Next" navigation buttons
- Ellipsis (...) for truncated page ranges when necessary
- Proper state management to reflect the current page in the URL

### Tests

- Verify that clicking on page numbers correctly updates the job listings.
- Ensure "Previous" and "Next" buttons function as expected, including edge cases (first and last pages).
- Test keyboard navigation through the pagination controls.
- Check that the component handles various total page number scenarios (few pages, many pages).
- Verify that the URL updates correctly when changing pages.

###  Tasks

- Design the pagination component based on the provided example and requirements.
- Develop the logic to generate the correct page numbers and ellipsis based on the current page and total pages.
- Implement click handlers for page numbers and navigation buttons.
- Integrate the pagination with the job listing data fetching mechanism.
- Add URL parameter management to reflect and control the current page.
- Utilize URL params for stage management
- Create animations or transitions for smoother page changes.
- Perform cross-browser testing and resolve any compatibility issues.

# LINKS

[`FIGMA LINK`](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=1553-45795&m=dev).

# IMAGES
image

### PR DESCRIPTION
**Pull request by skeby**
# Description

Implement pagination component

**Closes #96**

# Changes proposed

## What were you told to do?

Issue Title: [FEAT] External Dynamic Pages > Career Page > Implement Pagination
Implement pagination component and write tests to ensure the component works as expected.

## What did you do?

- Installed shacn pagination component (this also installed shadcn button component)
- Added a Pagination folder (contains the Pagination component and its test component) in the components common directory.
- Added a new css variable in globals.css for the border color of pagination

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
Screenshot (380)

image

image

### CODE REVIEW

File: src/components/common/Pagination/Pagination.test.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
+import { render, screen } from "@testing-library/react";
+import { expect } from "vitest";
```
Comment (oreoluwa212) [inline src/components/common/Pagination/Pagination.test.tsx:2]: remove this import skeby

**Review by oreoluwa212 — CHANGES_REQUESTED** [review]
remove the import skeby. vitest globals are already setup so you can take this out

File: src/components/common/Pagination/Pagination.test.tsx (hunk 1 above)
Comment (skeby) [reply src/components/common/Pagination/Pagination.test.tsx:2]: I don't know what you mean by "vitest globals", but when I removed the imports, I was seeing errors and the tests didn't run anymore
